dev-libs/libsecp256k1/libsecp256k1-0.0.0_pre20161213.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+inherit autotools eutils
+
+MyPN=secp256k1
+DESCRIPTION="Optimized C library for EC operations on curve secp256k1"
+HOMEPAGE="https://github.com/bitcoin/${MyPN}"
+COMMITHASH="8225239f490f79842a5a3b82ad6cc8aa11d5208e"
+SRC_URI="https://github.com/bitcoin/${MyPN}/archive/${COMMITHASH}.tar.gz -> ${PN}-v${PV}.tgz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~x86 ~amd64-linux ~x86-linux"
+IUSE="asm doc ecdh endomorphism experimental gmp java +recovery test test_openssl"
+
+REQUIRED_USE="
+	asm? ( || ( amd64 arm ) arm? ( experimental ) )
+	ecdh? ( experimental )
+	java? ( ecdh )
+	test_openssl? ( test )
+"
+RDEPEND="
+	gmp? ( dev-libs/gmp:0 )
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	java? ( virtual/jdk )
+	test_openssl? ( dev-libs/openssl:0 )
+"
+
+S="${WORKDIR}/${MyPN}-${COMMITHASH}"
+
+src_prepare() {
+	eautoreconf
+}
+
+src_configure() {
+	local asm_opt
+	if use asm; then
+		if use arm; then
+			asm_opt=arm
+		else
+			asm_opt=auto
+		fi
+	else
+		asm_opt=no
+	fi
+	econf \
+		--disable-benchmark \
+		$(use_enable experimental) \
+		$(use_enable java jni) \
+		$(use_enable test tests) \
+		$(use_enable test_openssl openssl-tests) \
+		$(use_enable ecdh module-ecdh) \
+		$(use_enable endomorphism)  \
+		--with-asm=$asm_opt \
+		--with-bignum=$(usex gmp gmp no) \
+		$(use_enable recovery module-recovery) \
+		--disable-static
+}
+
+src_install() {
+	if use doc; then
+		dodoc README.md
```

Comment:
in general, `README.md` is too innocuous to make doc-dependent, I would just install it unconditionally...